src/eradius_client.erl

```diff
@@ -435,8 +435,12 @@ store_upstream_servers(Server) ->
 store_radius_server_from_pool(Addr, Port, Retries) when is_tuple(Addr) and is_integer(Port) and is_integer(Retries) ->
     ets:insert(?MODULE, {{Addr, Port}, Retries, Retries});
 store_radius_server_from_pool(Addr, Port, Retries) when is_list(Addr) and is_integer(Port) and is_integer(Retries) ->
-    IP = get_ip(Addr),
-    ets:insert(?MODULE, {{IP, Port}, Retries, Retries});
+    try
+        IP = get_ip(Addr),
+        ets:insert(?MODULE, {{IP, Port}, Retries, Retries})
+    catch error:badarg ->
+        ?LOG(error, "Can't resolve hostname - ~p", [Addr])
+    end;
```

Comment:
I know that other places use the `try` around `get_ip` too, but that is just because `get_ip/1` is not really the right fit for those.
It would IMHO be far better to change the return of `get_ip/1` to `{ok, IP}|{error, Error}` and change the 3 places in this file that use it to handle that return properly.

Reply:
I thought about it but as get_ip is a part of public API I decided to not changeat least in this PR to not affect potential callers